### DISCUSSION TITLE
string validator does not return empty string when schema.string.nullableAndOptional() is used

### DISCUSSION THREAD

**Discussion by mtariqsajid** [adonisjs/core#4161] [Help]
## Prerequisites
- create a controller and validator and pass the request to validator

## Node.js and npm version
node: 18.15.0
npm: 9.5.0

## Sample Code (to reproduce the issue)
```js
import { schema, CustomMessages } from '@ioc:Adonis/Core/Validator'
import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'

export default class UserValidator {
  constructor(protected ctx: HttpContextContract) { }

  public schema = schema.create({
    id: schema.number(),
    name: schema.string.optional(),
    description: schema.string.nullableAndOptional(),
  })
}
```

when this validator invoke it does not return the field description if its empty pass from the request.
I want to validator to return empty string so i can update the user description field as empty string

**Comment by mtariqsajid** [comment]
anyone please respond @shreeve @allanfreitas @yann-yinn @mul14

**Comment by McSneaky** [comment]
So basically. You want there to be default values and in current case default value should be empty string?

I don't think validator supports default values, but you can add them manually after validation
```ts
const validated = await request.validate(UserValidator)
// If description isn't set, then give it default value
if (!validated.description) {
  validated.description = 'Foo bar and baz'
}
```

**Comment by mtariqsajid** [comment]
> So basically. You want there to be default values and in current case default value should be empty string?
> 
> I don't think validator supports default values, but you can add them manually after validation
> 
> ```ts
> const validated = await request.validate(UserValidator)
> // If description isn't set, then give it default value
> if (!validated.description) {
>   validated.description = 'Foo bar and baz'
> }
> ```

but this will become cumbersome, is their way to handle this if i have a lot of fields in model. and i might accidentally forgot about some model fields. how to keep it sync with model ? any suggestions . @McSneaky

**Comment by thetutlage** [comment]
If the user description is empty, then it must be represented as `null` in the database and not an empty string (for the sake of normalized data).

Therefore, I recommend marking the `description` column in your case as nullable and set the value to `null`.

**Reply by Sturez** [reply]
What if we need to make know the value sent by the callee. An empty string, null and undefined are three different values and they can have different meaning in a process. is there existing a way to address this use case?
